lm_eval/tasks/realtoxicityprompts/metric.py

```diff
@@ -26,6 +28,13 @@ def toxicity_perspective_api(references, predictions, **kwargs):
     url = f"https://commentanalyzer.googleapis.com/v1alpha1/comments:analyze?key={API_KEY}"
 
     scores = []
+
+
+    s = requests.Session()
+    retries = Retry(total=total_retries, backoff_factor=0.1, backoff_max=sleeping_time)
```

Comment:
```suggestion
    backoff_factor = sleeping_time / (2**(total_retries-1))
    retries = Retry(total=total_retries, backoff_factor=backoff_factor)
```

The max waiting time needed is 1 sec (`sleeping_time=1`). My solution would mean that we have a `backoff_factor = 1 / (2**4) = 0.0625`, which results in waiting `0.0625 0.125 0.25 0.5 1` seconds, after four tries we wait 0.9375s. Assuming time lost at other parts of the code, we could probably get a successful call with <`sleeping_time`. My Perspective API Key has a QPS of 25, which means a `sleeping_time=0.04`---not having `backoff_factor` depend on `sleeping_time` would result in unnecessary waiting times.

Regardless, `backoff_max` is not a variable we can set in `Retry()`, so we have to leave that out. We could make my solution less optimistic (already read `sleeping_time` at say the 3rd retry), but I think it should make the code overall more efficient than having a fixed `backoff_factor`.